spec/models/environment_spec.rb

```diff
@@ -16,17 +27,19 @@
   describe '#default_ref' do
     context 'when the environment has a default ref provided' do
       it 'returns that value' do
-        environment = Environment.new(default_ref: 'develop')
+        repo = Repository.with_name('acme-inc/api')
+        environment = Environment.new(default_ref: 'develop', repository: repo)
         expect(environment.default_ref).to eq 'develop'
       end
     end
 
     context 'when the environment does not have a default ref' do
       it 'returns the global default' do
-        environment = Environment.new(default_ref: '')
+        repo = Repository.with_name('acme-inc/api')
+        environment = Environment.new(default_ref: '', repository: repo)
         expect(environment.default_ref).to eq 'master'
 
-        environment = Environment.new
+        environment = Environment.new(repository: repo)
         expect(environment.default_ref).to eq 'master'
```

Comment:
The big change here is that we are updating this test so that an environment always has a repo attached. If we don't, our method fails. I can't think of a scenario where we have a lone environment not connected to a repo, but correct me if that's not the case.